midp/background.js

```diff
@@ -54,7 +54,19 @@ function backgroundCheck() {
   fgMidletNumber = (bgServer == 2) ? 1 : 2;
   fgMidletClass = MIDP.manifest["MIDlet-" + fgMidletNumber].split(",")[2];
 
-  DumbPipe.close(DumbPipe.open("backgroundCheck", {}));
+  if (!MIDlet.shouldStartBackgroundService ||
```

Comment:
> With it set to true, we'd only start in the background MIDlets that have the Nokia-MIDlet-bg-server property in their manifest (honoring the property seemed the best default option to me, while the heuristic is more of a hack to avoid starting MIDlets in the background when it isn't actually needed).

Ah, that makes sense, thanks for the explanation! Can you add this to the comment on the default version of the function? That way the next hacker who sees it will understand why we set the default this way.